frontend/src/stores/courses.js

```diff
@@ -0,0 +1,126 @@
+import { decorate, observable, action, autorun } from 'mobx';
+import mockCourse from './course-mock.json';
+import dateFns from 'date-fns';
+
+const isAvailableByTime = (courseItem) =>
+    // within 3 days from now
+    dateFns.differenceInDays(courseItem.startDate, new Date()) < 3 &&
+    // and must not be 1 hours before starting time
+    dateFns.differenceInHours(courseItem.startDate, new Date()) > 1;
+
+const pluckId = (obj) => {
+    const clone = [];
+    Object.keys(obj).forEach((key) =>
+        obj[key].forEach((item) => clone.push(item.id))
+    );
+    return clone;
+};
+
+const checkEvery5Sec = () => {
+    window.setTimeout(() => {
+        window.requestIdleCallback(() => {
+            this.checkAvailability();
+            checkEvery5Sec();
+        });
+    }, 5000);
+};
+
+class CourseStore {
+    courseList = [];
+    courseIdList = [];
+    isFetchingCourses = true;
+    useMockCourse = false;
+    filters = {
+        date: new Date(), // today
+    };
+    courseInFocus;
+
+    constructor() {
+        const checkEvery5Sec = () => {
```

Comment:
The (almost) same functionality is implemented as a function on top of the class.

Reply:
oh yeah wtf, Were I high or something???? THANK YOU